docs/proposals/diagnose-experience.md

```diff
@@ -0,0 +1,252 @@
+   "Accept": "application/vnd.docker.distribution.manifest.v2+json, application/vnd.docker.distribution.manifest.list.v2+json, application/vnd.oci.image.manifest.v1+json, application/vnd.oci.image.index.v1+json, application/vnd.oci.artifact.manifest.v1+json" 
+
+
+2024-08-02 23:56:03 <  msg=Response #0
+< Response Status: "401 Unauthorized"
+< Response headers:
+   "Content-Length": "73"
+   "X-Github-Request-Id": "9FC6:30019C:17C06:1C462:66AD0463"
+   "Content-Type": "application/json"
+   "Www-Authenticate": "Bearer realm=\"https://ghcr.io/token\",service=\"ghcr.io\",scope=\"repository:oras-project/oras:pull\""
+   "Date": "Fri, 02 Aug 2024 23:56:03 GMT"
+< Response body:
+{
+        "errors": [
+            {
+                "code": "<UNAUTHORIZED>",
+                "message": "<message describing condition>",
+                "detail": "<unstructured>"
+            },
+            ...
+        ]
+    }
+
+2024-08-02 23:56:02 > msg=Request #1
+> Request URL: "https://ghcr.io/token?scope=repository%3Aoras-project%2Foras%3Apull&service=ghcr.io"
+> Request method: "GET"
+> Request headers:
+   "User-Agent": "oras/1.2.0+Homebrew" 
+
+
+
+2024-08-02 23:56:03 < msg=Response #1
+< Response Status: "200 OK"
+< Response headers:
+   "Content-Type": "application/json"
+   "Docker-Distribution-Api-Version": "registry/2.0"
+   "Date": "Fri, 02 Aug 2024 16:08:05 GMT"
+   "Content-Length": "69"
+   "X-Github-Request-Id": "9FC6:30019C:17C0D:1C46C:66AD0464" 
+<  Response body:
+{
+  "schemaVersion": 2,
+  "mediaType": "application/vnd.oci.image.index.v1+json",
+  "manifests": [
+    {
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "digest": "sha256:42c524c48e0672568dbd2842d3a0cb34a415347145ee9fe1c8abaf65e7455b46",
+      "size": 1239,
+      "platform": {
+        "architecture": "amd64",
+        "os": "linux"
+      }
+    },
+    ···
+}
+
+
+```
+
+## Open Questions
+
+1. Should ORAS applies appropriate log levels to differentiate the log inforamtion? For example:
+
+- **Debug Level:** Reserve the `DEBUG` log level for detailed, technical information meant for developers.
+  - Example: `DEBUG: Parsed manifest with 3 layers. Digest: sha256:abcd1234`
+  
+- **Other Levels:** Use other log levels (Info, Warn, Error) to avoid cluttering debug logs with less vital information.
+  - Example (Info): `INFO: Successfully pushed artifact oras-demo:v1`
+  - Example (Error): `ERROR: Failed to push artifact oras-demo:v1 due to network timeout`
+
+2. Is it much more reasonable to output the user's environment details in `oras version` rather than in debug logs? The details about the environment where the tool is running, such as OS version and architecture, tool version, and environment variables (excluding sensitive data) would be super helpful to help ORAS developers and users to diagnose an issue.
+
+For example, the command and output will be as follows if output these information in `oras version`: 
```

Comment:
I would avoid this for now and have the users provide environment information themselves when reporting the issue.